docs/content/Theming/03_configuring_a_theme.md

```diff
@@ -132,3 +132,28 @@ angular.module('myApp', ['ngMaterial'])
 
 });
 </hljs>
+
+### Under the Hood
+
+Angular Material dynamically generates CSS for registered themes by injecting several
+`<script>` tags into the `<head>` section of the application at runtime. Here is how
+that process currently works:
+
+1. During the Angular Material build process, a function in `gulp/util.js` called
+[themeBuildStream](https://github.com/angular/material/blob/master/gulp/util.js#L187)
+compiles all `*-theme.scss` files from components into a single CSS string that is
+injected into the `material.core` module as a constant called `$MD_THEME_CSS`.
+
+1. At runtime, a function in `src/core/services/theming/theming.js` called
```

Comment:
Not sure if that is the case, but the default theme definitely produces 16 style tags in the head section. Here is a plunk with a build of the latest master commit: http://plnkr.co/edit/5W0q58xmvg76TSWmf32W?p=preview